host/rpi/rpi.go

```diff
@@ -128,17 +128,37 @@ func (d *driver) Init() (bool, error) {
 	//
 	// This code is not futureproof, it will error out on a Raspberry Pi 4
 	// whenever it comes out.
+	// Revision codes from: http://elinux.org/RPi_HardwareHistory
 	rev := distro.CPUInfo()["Revision"]
 	if i, err := strconv.ParseInt(rev, 16, 32); err == nil {
 		// Ignore the overclock bit.
 		i &= 0xFFFFFF
-		if i < 0x20 {
+		switch i {
+		case 0x0002, 0x0003, // B v1.0
+			0x0004, 0x0005, 0x0006, // B v2.0
+			0x0007, 0x0008, 0x0009, // A v2.0
+			0x000d, 0x000e, 0x000f, // B v2.0
+			0x0010,  // B+ v1.0
+			0x0011,  // Compute Module 1
+			0x0012,  // A+ v1.1
+			0x0013,  // B+ v1.2
+			0x0014,  // Compute Module 1
+			0x0015,  // A+ v1.1
+			0x90021, // A+ v1.1
```

Comment:
Ok I've read a bit more and I think this needs to be refactored to better represent the wide variety of boards.

What do you think about:  one variable to specify if the main GPIO header is 26 or 40 pins and a variable for presence of each secondary headers. These variables do not need to be exported. Then export below based on these variables. `Version` variable then should goes away since for practical purpose, Pi1 B+ and Pi2 B are similar ports wise and considering a Pi Zero W a Pi1 is very misleading, at this point it only means that the user cares about the type of CPU, which bcm283x is more appropriate to query this kind of information.

I found https://github.com/AndrewFromMelbourne/raspberry_pi_revision to be an interesting source of information.